### ISSUE TITLE
Need to find a replacement to Facebook SDK : package abandoned

### ISSUE THREAD

**Issue by stlgaits** [Myddleware/myddleware#618]
We should probably consider updating Facebook connector as the PHP SDK that we're using is abandoned and no replacement is suggested. This SDK requires a certain version of PHP, which is a hindrance to our other packages that could potentially be upgraded via composer if we removed the requirements linked to this SDK. Given that the development / update of this FB connector might take time, I think it would be probably wise to disable it for the time being and put it back up once the development is done.

For more info : https://packagist.org/packages/facebook/graph-sdk

**Comment by stlgaits** [comment]
The reason I am mentioning this is that if we manage to update all packages, we might then be PHP 8.0 compatible.

**Comment by Myddleware** [comment]
@EstelleMyddleware , I agree we can disable facebook connector on Myddleware 3 for now. We will update Facebook library later.

**Comment by rulotec** [comment]
Here's two options that might work:

* https://github.com/thephpleague/oauth2-facebook
* https://github.com/janu-software/facebook-php-sdk

**Comment by stlgaits** [comment]
@rulotec thanks for this, I will check them out and hopefully we can implement one of those instead ^^

**Comment by rulotec** [comment]
Thanks Estelle, I'd really like to use the latest version of myddleware with the FB connector.

One more question. If I install myddleware 3 and use php 7.4 (or less) and enable the fb connector through the database. Would that work?

**Comment by stlgaits** [comment]
@rulotec You can give it a go but as far as I'm aware I don't think it would given that this is the current SDK : https://packagist.org/packages/facebook/graph-sdk and if you have a look it doesn't go beyond PHP 7.0, is abandoned and hasn't been updated in years.

**Comment by rulotec** [comment]
I've been using myddleware 2 with php 7.2 for a long time using the facebook connector. I have now been using myddleware 2 with php 7.4 for about a week with the facebook connector without issues.

I would assume that if it's working correctly now, it would work in myddleware 3 as well if I also use php 7.4. Since the connector hasn't changed. Does that make sense?

**Comment by Myddleware** [comment]
Hi @rulotec ,
The facebook connector isn't enable in Myddleware 3. I see that we have made the connector compatible with Myddleware 3 but we haven't tested it.
So if you want, you can try it with Myddleware 3. You could test it and if you find some issue we could fix it. Then we could enable it in Myddleware 3 thanks to your tests ;).
Please let me know what you think.
